c2rust-analyze/src/c_void_casts.rs

```diff
@@ -158,25 +162,33 @@ pub struct CVoidCast<'tcx> {
 /// to their properly typed pointers,
 /// like [`CVoidCasts`], but in a single direction, meaning it represents either
 /// [`CVoidCastDirection::From`] or [`CVoidCastDirection::To`].
-#[derive(Default)]
-pub struct CVoidCastsUniDirectional<'tcx>(HashMap<CVoidPtr<'tcx>, Place<'tcx>>);
+#[derive(Default, Clone, Debug)]
+pub struct CVoidCastsUniDirectional<'tcx>(HashMap<Location, (CVoidPtr<'tcx>, Place<'tcx>)>);
```

Comment:
`(CVoidPtr<'tcx>, Place<'tcx>)` can just be `CVoidCast<'tcx>`.  They were only split before due to being the key and value of the `HashMap`.

Reply:
bc23fd71